src/models/syncRule.ts

```diff
@@ -79,16 +84,25 @@ export default class SyncRule {
         filters: SyncRuleTableFilters,
         pagination: TablePagination
     ): Promise<TableList> {
-        const { targetInstanceFilter = null } = filters || {};
+        const { targetInstanceFilter = null, enabledFilter = null, lastExecutedFilter = null } =
+            filters || {};
         const data = await getPaginatedData(d2, dataStoreKey, filters, pagination);
-        return targetInstanceFilter
-            ? {
-                  ...data,
-                  objects: _.filter(data.objects, e =>
-                      e.builder.targetInstances.includes(targetInstanceFilter)
-                  ),
-              }
-            : data;
+        return {
+            ...data,
+            objects: _(data.objects)
+                .filter(rule =>
+                    targetInstanceFilter
+                        ? rule.builder.targetInstances.includes(targetInstanceFilter)
+                        : true
+                )
+                .filter(rule => (enabledFilter ? rule.enabled === enabledFilter : true))
```

Comment:
No need to change anything, it's perfectly fine, but let me digress. In this pattern, to avoid the filtering when the guard condition is false, we could write: `.thru(rules => targetInstanceFilter ? rules.filter(...) : rules)`. Maybe someday we should simply create a mixin `_.filterIf(condition: boolean, predicate: (item: T) => boolean): T[]` :)